### PR TITLE
963: Bug during composer json generation

### PR DESCRIPTION
**Pull request by Iamwade**
1. Fixes magento/magento2-phpstorm-plugin#963

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java [hunk 1]
```diff
@@ -184,16 +184,15 @@ private String getDependenciesString(final List dependenciesList) {
                             composerJsonFile.getText()
                     );
                     final JSONObject jsonObject = (JSONObject) obj;
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java:186]: jsonObject can be null as well.

File: src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java [hunk 2]
```diff
@@ -184,16 +184,15 @@ private String getDependenciesString(final List dependenciesList) {
...
-                    final String versionJsonElement = jsonObject.get("version").toString();
+                    final String versionJsonElement = jsonObject.get("version") == null
+                            ? "*" : jsonObject.get("version").toString();
                     final String nameJsonElement = jsonObject.get("name").toString();
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java:189]: jsonObject.get("name") can be null as well.

File: src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java [hunk 3]
```diff
@@ -184,16 +184,15 @@ private String getDependenciesString(final List dependenciesList) {
...
 
-                    if (versionJsonElement != null) {
-                        version = versionJsonElement;
-                        final int minorVersionSeparator = version.lastIndexOf('.');
-                        version = new StringBuilder(version)
-                                .replace(minorVersionSeparator + 1, version.length(),"*")
-                                .toString();
-                    }
+                    version = versionJsonElement;
+                    final int minorVersionSeparator = version.lastIndexOf('.');
```
Comment (bohdan-harniuk) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/ModuleComposerJsonGenerator.java:192]: We should not do this if there is no version specified.

**Review by bohdan-harniuk — CHANGES_REQUESTED** [review]
Hello, @Iamwade!

Please, check my suggestions.

**Review by bohdan-harniuk — APPROVED** [review]
Hello, @Iamwade!

Looks better than before!
Thank you for your work here!

@eduard13, what do you think?

Regards,

**Review by eduard13 — APPROVED** [review]
Looks good 👍

**Pull request merged**